tests/test_diaCalculationPlugins.py

```diff
@@ -872,6 +880,76 @@ def testCalculate(self):
                                np.sqrt(1 / (n_sources - 1)))
 
 
+class TestMultiLombScarglePeriodogram(unittest.TestCase):
+
+    def testCalculate(self):
+        """Test Mulitband Lomb Scargle Periodogram."""
+        n_sources = 10
+        objId = 0
+
+        # Create synthetic multi-band data
+        times = np.linspace(0, 2*np.pi, n_sources)
+        fluxes = np.sin(0.3 * times)
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "band": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "midpointMjdTai": times,
+                  "psfFlux": fluxes,
+                  "psfFluxErr": np.zeros(n_sources)})
+
+        plug = LombScarglePeriodogramMulti(LombScarglePeriodogramMultiConfig(),
+                                           "ap_lombScarglePeriodogramMulti",
+                                           None)
+
+        run_multi_plugin(diaObjects, diaSources, "u", plug)
+        self.assertAlmostEqual(diaObjects.at[objId, "multi_period"],
+                               31.41592653589793)
+
+
+class TestLombScarglePeriodogram(unittest.TestCase):
+
+    def testCalculate(self):
+        """Test Lomb Scargle Periodogram."""
+        n_sources = 10
+        objId = 0
+
+        # Test on periodic scienceFlux.
+        times = np.linspace(0, 2*np.pi, n_sources)
+        fluxes = np.sin(0.3 * times)
+        diaObjects = pd.DataFrame({"diaObjectId": [objId]})
+        diaSources = pd.DataFrame(
+            data={"diaObjectId": n_sources * [objId],
+                  "band": n_sources * ["u"],
+                  "diaSourceId": np.arange(n_sources, dtype=int),
+                  "midpointMjdTai": times,
+                  "psfFlux": fluxes,
+                  "psfFluxErr": np.zeros(n_sources)})
+
+        plug = LombScarglePeriodogram(LombScarglePeriodogramConfig(),
+                                      "ap_lombScarglePeriodogram",
+                                      None)
+
+        run_multi_plugin(diaObjects, diaSources, "u", plug)
+        self.assertAlmostEqual(diaObjects.at[objId, "u_period"],
```

Comment:
Please test both period and power, and note where these expected values came from (even if you just calculated them by hand because they're "trivial" given the structure of the above data).

Reply:
Added.